### PR TITLE
add some basic stats, and test case

### PR DESCRIPTION
**Pull request by yifan-gu**
1,
I know the stats name is ugly...
some name strategy is under consideration, like xxxErrors, Successfulxxxs,

2,
Concurrent correctness has not been tested

3,
I can not find a way to use relative path for the "error package", so I used absolute path, ("github.com/xiangli-cmu/etcd/error")

4,
To avoid infinite loop when call json.Marshal, now the Watcher under the FileSystem has a member of a Stats pointer, which points to the same Stats owned by FilesSystem

### CODE REVIEW

File: file_system/stats_test.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+	"testing"
+	"time"
+	//"fmt"
+)
+
+func TestBasicStats(t *testing.T) {
+	fs := New()
+	keys := GenKeys(rand.Intn(100), 5)
+
+	i := uint64(0)
```
Comment (xiang90) [inline file_system/stats_test.go:14]: You do not to init var to zero in go. It will help you to do that.

File: file_system/stats.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+func (e *EtcdStats) TotalWrites() uint64 {
+	return e.SetsHit + e.SetsMiss +
+		e.DeletesHit + e.DeletesMiss +
+		e.UpdatesHit + e.UpdatesMiss +
+		e.TestAndSetsHit + e.TestAndSetsMiss
+}
+
+func (e *EtcdStats) IncStats(field int) {
+	if field >= 200 {
+		e.rwlock.Lock()
```
Comment (xiang90) [inline file_system/stats.go:96]: Why not use sync/atomic package? I do not need a lock here.

**Comment by xiang90** [comment]
@evan-gu To avoid look in `json`, you should do like 

``` go
// Field is ignored by this package.
Field int `json:"-"`
```

to ignore the field, which will cause the loop.
But you need to make sure to recover the field when restart from snapshot.

File: file_system/stats_test.go (hunk 1 above)
Comment (yifan-gu) [inline file_system/stats_test.go:14]: I need uint64, which is not the default

File: file_system/stats.go (hunk 2 above)
Comment (yifan-gu) [inline file_system/stats.go:96]: well i walked through the source code of sync/atomic, it implements the the atomic add by using a mutex lock.
And in this case, not every operation should be blocked by mutex, a rwlock is better

File: file_system/stats_test.go (hunk 1 above)
Comment (xiang90) [inline file_system/stats_test.go:14]:
```
var (
  GetHits uint64
  ....
)
```

File: file_system/stats_test.go [hunk 3]
```diff
@@ -0,0 +1,221 @@
+package fileSystem
+
+import (
+	"math/rand"
+	"testing"
+	"time"
+	//"fmt"
```
Comment (xiang90) [inline file_system/stats_test.go:7]: remove commented codes

**Pull request closed without merging**